### DISCUSSION TITLE
downloading all data blobs to create a long term data availability solution

### DISCUSSION THREAD

**Discussion by sudeepdino008** [erigontech/erigon#7916] [General]
Hi, we're working on a data availability solution for ethereum (in the face of state expiry or protodanksharding/danksharding). My question is specific to protodanksharding/danksharding -- will a geth client (in some mode) be able to get ALL the blobs currently in the network? Given the there's data sharding, I suppose certain nodes which don't belong in the relevant subnet, won't be able to download the blobs by default? 
I want to get some pointers on the feasability of this, or how we might be able to do this. What are your thoughts?

**Comment by AskAlexSharov** [comment]
Ethereum doesn’t have sharding and expiry yet. So, by default erigon just have all the data (archive node).

**Reply by sudeepdino008** [reply]
I did mean to ask it as how it might be implemented "going ahead", since proto-danksharding one of the next major items on the list for rollup scalability. I suppose it's currently at the EIP/protocol design level, and the Eth clients are not involved right now.
